zulipterminal/ui_tools/views.py

```diff
@@ -187,10 +187,12 @@ def read_message(self, index: int=-1) -> None:
         if msg_w is None:
             return
         self.update_search_box_narrow(msg_w.original_widget)
+        # If this the last message in the view and focus is set on this message
+        # then read the message.
+        last_message_focused = (curr_pos == len(self.log) - 1)
         # Only allow reading a message when middle column is
-        # in focus. Changing the contents of self.log also changes
-        # focus automatically.
-        if view.body.focus_col != 1:
+        # in focus.
+        if view.body.focus_col != 1 and not last_message_focused:
```

Comment:
If I understand what you're trying to do here (and by de Morgan's), would this be clearer as:
```python
not(focus_col == 1 or last_message_focused)
```
If so, are you trying to enable the last message to be read even if the (column) focus is not on the message column?
In what circumstances does this happen?

Reply:
I too had the same doubt here regarding when the second commit here would do?

Reply:
It happens when the user has only one unread message and open's zt. 
This will be the behavior without this change:
![Screenshot (1)](https://user-images.githubusercontent.com/25124304/60391969-f7798480-9b17-11e9-8577-4ca9b25ba91e.png)
![Screenshot (2)](https://user-images.githubusercontent.com/25124304/60391970-f9dbde80-9b17-11e9-896c-e944c3ba1dbb.png)

Reply:
Ah thanks for the clarification @amanagr. it would help to have that line in commit description, if possible.